From 91aa70c6320461ae65600dce954b5ffb905b7039 Mon Sep 17 00:00:00 2001
From: Eli Schwartz <eschwartz93@gmail.com>
Date: Mon, 9 Apr 2018 12:36:35 -0400
Subject: [PATCH] Do not vendor the pugixml library

This prevents trying to co-install pugixml with existing system
libraries, and also fixes the includes for pugixml.hpp to actually use
the include path added by find_package/add_subdirectory instead of a
hardcoded relative path.
---
 CMakeLists.txt             |     3 +-
 pugixml/CMakeLists.txt     |    80 -
 pugixml/src/pugiconfig.hpp |    74 -
 pugixml/src/pugixml.cpp    | 12622 -------------------------------------------
 pugixml/src/pugixml.hpp    |  1434 -----
 src/MeshData.cpp           |     7 +-
 src/Scene.cpp              |     2 +-
 src/SceneNode.cpp          |     3 +-
 src/ThreeMFParser.h        |     4 +-
 9 files changed, 7 insertions(+), 14222 deletions(-)
 delete mode 100644 pugixml/CMakeLists.txt
 delete mode 100644 pugixml/src/pugiconfig.hpp
 delete mode 100644 pugixml/src/pugixml.cpp
 delete mode 100644 pugixml/src/pugixml.hpp

diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt    2019-06-21 01:28:05.044932934 +0300
+++ b/CMakeLists.txt    2019-06-21 01:26:38.837837677 +0300
@@ -14,7 +14,7 @@
     find_package(Threads QUIET)
 endif()
 
-add_subdirectory(pugixml)
+find_package(pugixml REQUIRED)
 
 if(BUILD_PYTHON)
     list(APPEND CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake)
diff -Naur a/src/MeshData.cpp b/src/MeshData.cpp
--- a/src/MeshData.cpp  2019-04-17 17:58:53.000000000 +0300
+++ b/src/MeshData.cpp  2019-06-21 01:28:53.630168506 +0300
@@ -17,7 +17,7 @@
  */
 
 #include "MeshData.h"
-#include "../pugixml/src/pugixml.hpp"
+#include <pugixml.hpp>
 #include <cstring>
 #include <iostream>
 #include <stdexcept> //For std::runtime_error.
diff -Naur a/src/Scene.cpp b/src/Scene.cpp
--- a/src/Scene.cpp     2019-06-21 01:29:21.012610430 +0300
+++ b/src/Scene.cpp     2019-06-21 01:26:38.837837677 +0300
@@ -17,7 +17,7 @@
  */
 
 #include "Scene.h"
-#include "../pugixml/src/pugixml.hpp"
+#include <pugixml.hpp>
 #include <iostream>
 #include <string>
 using namespace Savitar;
diff -Naur a/src/SceneNode.cpp b/src/SceneNode.cpp
--- a/src/SceneNode.cpp 2019-06-21 01:29:28.964157975 +0300
+++ b/src/SceneNode.cpp 2019-06-21 01:26:38.837837677 +0300
@@ -17,7 +17,7 @@
  */
 
 #include "SceneNode.h"
-#include "../pugixml/src/pugixml.hpp"
+#include <pugixml.hpp>
 #include <iostream>
 using namespace Savitar;
 
diff -Naur a/src/ThreeMFParser.h b/src/ThreeMFParser.h
--- a/src/ThreeMFParser.h       2019-06-21 01:29:36.303740339 +0300
+++ b/src/ThreeMFParser.h       2019-06-21 01:26:38.837837677 +0300
@@ -21,7 +21,7 @@
 
 #include "SavitarExport.h"
 #include "SceneNode.h"
-#include "../pugixml/src/pugixml.hpp"
+#include <pugixml.hpp>
 
 #include <string>
 namespace Savitar
-#endif
\ No newline at end of file
+#endif
